uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/test/java/org/uberfire/ext/wires/core/grids/client/widget/grid/renderers/grids/impl/BaseGridRendererNonSelectionLayerTest.java

```diff
@@ -112,7 +117,53 @@ public void checkSelectedCellsNotClippedByHeader() {
                                    1,
                                    3,
                                    0,
-                                   2);
+                                   2,
+                                   column.getWidth());
+    }
+
+    @Test
+    public void checkSelectedCellsWithHiddenColumnSingleColumn() {
+        column.setVisible(false);
+        checkRenderedSelectedCells(0,
+                                   0,
+                                   1,
+                                   3,
+                                   0,
+                                   2,
+                                   0.0);
+    }
+
+    @Test
+    public void checkSelectedCellsWithHiddenColumnMultipleColumns() {
+        final GridColumn<String> column2 = makeGridColumn(200.0);
+        this.model.appendColumn(column2);
+        setupSelectionContext();
+
+        checkRenderedSelectedCells(0,
+                                   0,
+                                   2,
+                                   1,
+                                   0,
+                                   0,
+                                   column.getWidth() + column2.getWidth());
+    }
+
+    @Test
+    public void checkSelectedCellsWithHiddenColumnMultipleColumnsFirstHidden() {
+        final GridColumn<String> column2 = new StringPopupColumn(new BaseHeaderMetaData("title"),
+                                                                 columnRenderer,
+                                                                 200.0);
```

Comment:
Could we use `makeGridColumn` instead of calling `StringPopupColumn`?

Reply:
@jomarko Good spot.. I obviously didn't change that method after I decided to have the ```makeGridColumn(..)``` method. Updated in new commit.